src/registry.ts

```diff
@@ -337,3 +315,40 @@ export function validateRegistryEntry(name: string, value: unknown, valueKind: s
   validateString(`${name}.data`, (value as RegistryEntry).data, `${valueKind} field`);
   validateBigint(`${name}.revision`, (value as RegistryEntry).revision, `${valueKind} field`);
 }
+
+/**
+ * Handles error responses returned in getEntry.
+ *
+ * @param err - The Axios error.
+ * @returns - An empty signed registry entry if the status code is 404.
+ * @throws - Will throw if the status code is not 404.
+ */
+function handleGetEntryErrResponse(err: AxiosError): SignedRegistryEntry {
```

Comment:
I think we want to throw a custom error here that signals to the SDK user he's being ratelimited.

Reply:
It returns this:
```
Error response did not contained expected fields 'data.message'. Status code: 429. Full error: Error: Request failed with status code 429
```

Which is good enough, but I don't think it's the most user friendly error message.